packages/guides-restructured-text/src/RestructuredText/Parser/Productions/LinkRule.php

```diff
@@ -50,11 +50,11 @@ public function apply(BlockContext $blockContext, CompoundNode|null $on = null):
         $node = null;
         if ($link->getType() === LinkParser::TYPE_ANCHOR) {
             $node = new AnchorNode($link->getName());
```

Comment:
if we return $node here, we can keep the else statement as before.

Reply:
I think with if and else it is much better to see there are alternative paths.I don't like early returns except for Handling of Edge cases / Errors